react-js/react-hooks.md

```diff
@@ -2,89 +2,284 @@
+- [Pure functions in React](https://dev.to/keevcodes/pure-functions-in-react-2o7n)
+- [Intro to React hooks](https://reactjs.org/docs/hooks-intro.html)
+- [React hooks and state](https://reactjs.org/docs/hooks-state.html)
 
 ### Lesson
 
-Here's text about introducing something and how it works.
-
-- [Be sure to link the slideshow, video,](https://google.com)
-- [or other materials you expect them to follow as part of the lesson.](https://google.com)
+#### What are pure functions?
+ Look at the  functions in the [pure functions article](https://dev.to/keevcodes/pure-functions-in-react-2o7n) (also copied below). Why is `multiply` *not* a pure function? Why is `multiplyNumber` a pure function?
+ ```const globalNumber = 4;
+const multiply = (x) => {
+  return globalNumber *= x 
+}
+```  
+
+ ```
+const multiplyNumber = (x) => {
+  return x * 2;
+}
+```
+Pure components, also called (Stateless)Functional Component, or *(S)FC* in React, are pure functions - they are components that do not store any internal state, and are pure with respect to their props. For example,
+ ```
+const HeadlineComponent = props => (<h1>{props.headline}</h1>)
+``` 
+
+#### Why were hooks introduced?
+Read [Intro to react hooks](https://reactjs.org/docs/hooks-intro.html) (5-10 minutes). 
+
+ Note:  React hooks were introduced in version 16.8. Previous alternatives to hooks include [render props](https://reactjs.org/docs/render-props.html),
+ [higher-order components](https://reactjs.org/docs/higher-order-components.html) such as the HOC used to connect a component to [the redux store](https://code.likeagirl.io/how-to-create-a-connected-hoc-with-react-and-redux-8631f7b1497). 
+ Right now you don't have to understand all of these alternatives in depth. The main point to note is there were a lot of solutions for the same problem.Hooks introduce *one* concept that solves these problems. 
+
+#### How do I use hooks?
+1. Read [Hooks Overview](https://reactjs.org/docs/hooks-overview.html)
+2. Walk through the examples on [Using the State Hook](https://reactjs.org/docs/hooks-state.html) to understand the state hook and its Class equivalent (in the counter `Example` component ).  Use [this codepen]([https://codepen.io/dpenny/pen/RwPPVqY](https://codepen.io/dpenny/pen/RwPPVqY)), to play around with these two ways of creating the same functionality. 
+
+*Discuss with a partner:*
+How do hooks achieve the same logic as the class component? Can you see how the code with hooks is cleaner?
 
-Build on the first information. Have apprentices guess things, do an activity, etc.
-
-Make sure to mention these things:
-
-- Things
-	- This is a sub-thing
-- More things
-- Even more things
-- Even more things
+### Guided Practice
+**Goal:  understand how we would use React hooks to implement a simple to-do list.**
+
+A user should be able to:
+-  Add items (via text input field)
+-  Press a button 'All done' which will clear all todo items, and a window alert will say 'Good job!'
```

Comment:
This isn't a valid requirement anymore, right? The status is displayed on the page now?